tests/testthat/test-duplicates.R

```diff
@@ -106,3 +96,67 @@ test_that("errors when input argument is not a data.frame", {
   expect_error(duplicates(NULL), "Data.frame must be a data.frame.")
   expect_error(duplicates(NA), "Data.frame must be a data.frame.")
 })
+
+test_that("preserves groups", {
+  data <- tibble::tibble(
+    X = c(1, 2, 2, 3, 3, 4, 4),
+    Y = c(11, 12, 13, 14, 14, 15, 15),
+    a = c("red", "orange", "yellow", "green", "green", "blue", "blue"),
+    b = c("white", "white", "white", "white", "white", "white", "white")
+  )
+  data <- dplyr::group_by(data, a, b)
+  
+  data_dup <- tibble::tibble(
+    X = c(3, 3, 4, 4),
+    Y = c(14, 14, 15, 15),
+    a = c("green", "green", "blue", "blue"),
+    b = c("white", "white", "white", "white")
+  )
+  data_dup <- dplyr::group_by(data_dup, a, b)
+  
+  expect_identical(data_dup, duplicates(data))
```

Comment:
I would apply the `duplicates()` function outside of the expect call if possible as it makes the code more readable.

Reply:
@aylapear, I did it this way to be consistent with the previously written tests. Would you like me to change all of the tests, just the tests that I wrote, or keep it the way it is now?

Reply:
Just the tests you wrote for now.